web_services/foxx/brightid/db.js

```diff
@@ -567,9 +575,11 @@ function getLastContextIds(coll, appKey) {
 }
 
 function userVerifications(user) {
-  const verifications = verificationsColl.byExample({
-    user
-  }).toArray();
+  const hashes = variablesColl.document('VERIFICATIONS_HASHES').hashes;
+  const snapshotPeriod = hashes[1]['block'] - hashes[0]['block']
+  const lastBlock = variablesColl.document('LAST_BLOCK').value;
+  const block = lastBlock - (lastBlock % snapshotPeriod) - snapshotPeriod
```

Comment:
This appear to be getting the second value of `hashes[]['block']` that's less than `lastBlock`--in other words, the block that marks the end of the second-to-last snapshot that this node has available.

This would be easier to understand if it were computed as a loop rather than a remainder and subtraction, or at least if there were a comment.

Reply:
hmmm
We are getting block numbers not timestamp here. `snapshotPeriod` become 240 based on current configuration.
So if last block processed by consensus service is `n * 240 + x` (where `x` is lower than 240), we should have verifications for block `(n-1) * 240` and probably also block `n * 240` (if having enough time) in database. We should query and use verification calculated for block `(n-1) * 240` in verification endpoints because all nodes had enough time (at least one `snapshotPeriod`) to calculate that.

I'm not sure what you mean by computing it as a loop and if we have same understanding about what we are doing here.
If you think code is fine, I can add a comment to provide more details.

Reply:
I understand you, but apparently you don't understand me.  What you just described is what I figured out from looking at the code.  I'm saying the code isn't easy to understand, so no I don't think it's fine.